lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_playbook_method.rb

```diff
@@ -0,0 +1,112 @@
+module MiqAeEngine
+  class MiqAePlaybookMethod
+    PLAYBOOK_CLASS = ManageIQ::Providers::EmbeddedAutomationManager::ConfigurationScriptPayload
+
+    def initialize(aem, obj, inputs)
+      @workspace = obj.workspace
+      @inputs    = inputs
+      @aem       = aem
+    end
+
+    def run
+      if @workspace.persist_state_hash[method_key].present?
+        task_id = @workspace.persist_state_hash[method_key]
+        @aw = AutomateWorkspace.find_by(:guid => @workspace.persist_state_hash['automate_workspace_guid'])
+        check_task_status(task_id)
+      else
+        execute
+      end
+    end
+
+    private
+
+    def execute
+      @aw = AutomateWorkspace.create(:input  => serialize_workspace,
+                                     :user   => @workspace.ae_user,
+                                     :tenant => @workspace.ae_user.current_tenant)
+      playbook_options = build_options_hash
+      $miq_ae_logger.info("Playbook Method passing options to runner: #{playbook_options}")
+      begin
+        task_id = PLAYBOOK_CLASS.run(playbook_options)
+      rescue => err
+        $miq_ae_logger.error("Playbook Method Ended with error #{err.message}")
+        reset
+        raise MiqAeException::AbortInstantiation, err.message
+      end
+
+      @workspace.root['ae_state_started'].blank? ? wait_for_method(task_id) : check_task_status(task_id)
+    end
+
+    def process_result
+      @aw.reload
+      @workspace.update_workspace(@aw.output)
+      reset
+    end
+
+    def reset
+      @workspace.persist_state_hash.delete('automate_workspace_guid')
+      @workspace.persist_state_hash.delete(method_key)
+      @aw.delete
+    end
+
+    def wait_for_method(task_id)
+      task = MiqTask.wait_for_taskid(task_id)
+      raise MiqAeException::Error, task.message unless task.status == "Ok"
+      process_result
+    ensure
+      reset
+    end
+
+    def check_task_status(task_id)
+      task = MiqTask.find(task_id)
+      raise MiqAeException::Error, "Task id #{task_id} not found" unless task
+      if task.state == MiqTask::STATE_FINISHED
+        if task.status == MiqTask::STATUS_OK
+          @workspace.root['ae_result'] = 'ok'
+          process_result
+        else
+          @workspace.root['ae_result'] = 'error'
+          reset
+          raise MiqAeException::Error, task.message
+        end
+      else
+        @workspace.root['ae_result'] = 'retry'
+        @workspace.root['ae_retry_interval'] = 1.minute
+        @workspace.persist_state_hash['automate_workspace_guid'] = @aw.guid
+        @workspace.persist_state_hash[method_key] = task_id
```

Comment:
Also for readability can we move this "retry" logic into a separate method.  If we end up supporting a way to set the interval this logic will likely become more complex and would be better in a separate method.